Source/JavaScriptCore/runtime/ModuleProgramExecutable.cpp

```diff
@@ -55,8 +55,7 @@ UnlinkedModuleProgramCodeBlock* ModuleProgramExecutable::getUnlinkedCodeBlock(JS
     OptionSet<CodeGenerationMode> codeGenerationMode = globalObject->defaultCodeGenerationMode();
     unlinkedModuleProgramCode = vm.codeCache()->getUnlinkedModuleProgramCodeBlock(vm, this, source(), codeGenerationMode, error);
 
-    if (globalObject->hasDebugger())
-        globalObject->debugger()->sourceParsed(globalObject, source().provider(), error.line(), error.message());
+    globalObject->sourceParsed(source().provider(), error.line(), error.message());
```

Comment:
Always asking to the hashtable is too costly. You need to come up with an idea not doing that when debugger is disabled.